fragmenter/fragment.py

```diff
@@ -824,1078 +658,459 @@ class WBOFragmenter(Fragmenter):
-            fn = resource_filename('fragmenter', os.path.join('data', 'fgroup_smarts.yml'))
-            with open(fn, 'r') as f:
-                functional_groups = yaml.safe_load(f)
-        self._options['functional_groups'] = functional_groups
-        self._tag_functional_groups(functional_groups)
+            functional_groups = get_fgroup_smarts()
+
+        super().__init__(molecule, functional_groups)
 
         self.rotors_wbo = {}
-        self.ring_systems = {}
-        self.fragments = {} # Fragments from fragmentation scheme for each rotatable bond
-        self._fragments = {} # Used internally for depiction
 
+        self.verbose = verbose
+        self.threshold = None
 
-    def fragment(self,  threshold=0.03, keep_non_rotor_ring_substituents=False, **kwargs):
+    def fragment(
+        self, threshold=0.03, keep_non_rotor_ring_substituents=False, **kwargs
+    ):
         """
         Fragment molecules using the Wiberg Bond Order as a surrogate
 
         Parameters
         ----------
+        threshold : float, optional, default 0.01
+            The threshold for the central bond WBO. If the fragment WBO is below this
+            threshold, fragmenter will grow out the fragment one bond at a time via the
+            path specified by the heuristic option
         keep_non_rotor_ring_substituents: bool
-            If True, will always keep all non rotor substiuents on ring. If False, will only add them
-            if they are ortho to rotatable bond or if it's needed for WBO to be within the threshold
+            If True, will always keep all non rotor substiuents on ring. If False, will
+            only add them if they are ortho to rotatable bond or if it's needed for WBO
+            to be within the threshold
         **heuristic : str, optional, default 'path_length'
-            The path fragmenter should take when fragment needs to be grown out. The other option is
-            'wbo'
-        **threshold : float, optional, default 0.01
-            The threshold for the central bond WBO. If the fragment WBO is below this threshold, fragmenter
-            will grow out the fragment one bond at a time via the path specified by the heuristic option
-
+            The path fragmenter should take when fragment needs to be grown out. The
+            other option is 'wbo'
         """
         # Capture options used
-        self._options['keep_non_rotor_ring_substituents'] = keep_non_rotor_ring_substituents
-        if 'threshold' not in self._options:
-            self._options['threshold'] = threshold
+        self._options[
+            "keep_non_rotor_ring_substituents"
+        ] = keep_non_rotor_ring_substituents
+        if "threshold" not in self._options:
+            self._options["threshold"] = threshold
 
         # Add threshold as attribute because it is used in more than one function
-        setattr(self, 'threshold', threshold)
+        setattr(self, "threshold", threshold)
```

Comment:
Is this different from `self.threshold = threshold`?

Reply:
No, good catch - this was just something I missed!

Reply:
Resolved in 014af99